sdk/core/Microsoft.Azure.Core.Spatial/src/Serialization/GeographyConverter.cs

```diff
@@ -0,0 +1,107 @@
+        private static readonly JsonEncodedText s_TypePropertyNameBytes = JsonEncodedText.Encode(TypePropertyName);
+
+        /// <inheritdoc/>
+        public override bool CanConvert(Type typeToConvert) =>
+            // BUGBUG #15506: Check for all Geography derivatives.
+            typeof(GeographyPoint).IsAssignableFrom(typeToConvert);
+
+        /// <inheritdoc/>
+        public override Geography Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            string type = default;
+            double? longitude = default;
+            double? latitude = default;
+
+            reader.Expect(JsonTokenType.StartObject);
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                reader.Expect(JsonTokenType.PropertyName);
+                string propertyName = reader.GetString();
+
+                reader.Read();
+                if (string.Equals(TypePropertyName, propertyName, StringComparison.Ordinal))
+                {
+                    reader.Expect(JsonTokenType.String);
+                    type = reader.GetString();
+                }
+                else if (string.Equals(CoordinatesPropertyName, propertyName, StringComparison.Ordinal))
+                {
+                    reader.Expect(JsonTokenType.StartArray);
+
+                    // Longitude
+                    reader.Read();
+                    reader.Expect(JsonTokenType.Number);
+                    longitude = reader.GetDouble();
+
+                    // Latitude
+                    reader.Read();
+                    reader.Expect(JsonTokenType.Number);
+                    latitude = reader.GetDouble();
+
+                    // Skip the rest.
+                    do
+                    {
+                        reader.Read();
+                    } while (reader.TokenType != JsonTokenType.EndArray);
+                }
+                else
+                {
+                    reader.Skip();
+                }
+            }
+
+            if (!string.Equals(PointTypeName, type, StringComparison.Ordinal))
+            {
+                throw new JsonException($"Deserialization of {nameof(GeographyPoint)} failed. Expected geographic type: '{PointTypeName}'.");
+            }
+
+            if (!longitude.HasValue || !latitude.HasValue)
+            {
+                throw new JsonException($"Deserialization of {nameof(GeographyPoint)} failed. Expected both longitude and latitude.");
+            }
+
+            return GeographyPoint.Create(latitude.Value, longitude.Value);
+        }
+
+        /// <inheritdoc/>
+        public override void Write(Utf8JsonWriter writer, Geography value, JsonSerializerOptions options)
+        {
+            if (value is GeographyPoint point)
```

Comment:
What about altitude?

Reply:
We didn't read or write it before. That's why I opened #15483 to track.